### ISSUE TITLE
Equality fails

### ISSUE THREAD

**Issue by nopara73** [MetacoSA/NBitcoin#252]
As @andzsy reported here: https://github.com/ProgrammingBlockchain/ProgrammingBlockchain/issues/51
```cs
var privateKey = new Key();
var bitcoinSecret = privateKey.GetWif(Network.Main);
var samePrivateKey = bitcoinSecret.PrivateKey;
Console.WriteLine(samePrivateKey == privateKey);  // Should  be true, but the result is false
```

**Comment by andzsy** [comment]
This is happening on NBitcoin 2.1.5.14, btw.

**Comment by knocte** [comment]
> This is happening on NBitcoin 2.1.5.14, btw.

Has this ever worked? is it a regression?

**Comment by andzsy** [comment]
I'm not sure, I'm just going through the Programming Blockchain tutorial and the book's output differs from the real-life one.

**Comment by NicolasDorier** [comment]
fixed in https://github.com/MetacoSA/NBitcoin/commit/2ca45cfd8140d1e8fb258a18704ce336dfb4de8c . We are comparing correctly `PubKey` so comparing `Key` should be supported as well. (fixed for 3.0.2.24)
